flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/strategy/EagerSchedulingStrategy.java

```diff
@@ -72,27 +69,24 @@ public void onPartitionConsumable(ExecutionVertexID executionVertexId, ResultPar
 		// Will not react to these notifications.
 	}
 
-	private void allocateSlotsAndDeploy(final Set<ExecutionVertexID> verticesToDeploy) {
+	private void allocateSlotsAndDeploy(final List<ExecutionVertexID> verticesToDeploy) {
 		final List<ExecutionVertexDeploymentOption> executionVertexDeploymentOptions =
 				createExecutionVertexDeploymentOptions(verticesToDeploy);
 		schedulerOperations.allocateSlotsAndDeploy(executionVertexDeploymentOptions);
 	}
 
-	private Set<ExecutionVertexID> getAllVerticesFromTopology() {
-		return StreamSupport
-				.stream(schedulingTopology.getVertices().spliterator(), false)
+	private List<ExecutionVertexID> getAllVerticesFromTopology() {
+		return IterableUtils.toStream(schedulingTopology.getVertices())
 				.map(SchedulingExecutionVertex::getId)
-				.collect(Collectors.toSet());
+				.collect(Collectors.toList());
 	}
 
 	private List<ExecutionVertexDeploymentOption> createExecutionVertexDeploymentOptions(
-			final Collection<ExecutionVertexID> vertices) {
-		List<ExecutionVertexDeploymentOption> executionVertexDeploymentOptions = new ArrayList<>(vertices.size());
-		for (ExecutionVertexID executionVertexID : vertices) {
-			executionVertexDeploymentOptions.add(
-					new ExecutionVertexDeploymentOption(executionVertexID, deploymentOption));
-		}
-		return executionVertexDeploymentOptions;
+			final List<ExecutionVertexID> vertices) {
+
+		return vertices.stream()
```

Comment:
ll. 87 to 89 are changes that are strictly speaking not necessary

Reply:
yes. It is dropped.